ordnancesurvey/places.py

```diff
@@ -5,18 +5,42 @@
 from dotenv import load_dotenv
 load_dotenv()
 
-
 def get_places_from_search(search):
     """Get places results from search term"""
 
     api_url = "https://api.os.uk/search/places/v1/find?"
     params = {"query": search, 'key': os.environ.get(
         "os_api_key"), 'maxresults': '30'}
     response = requests.get(api_url + urllib.parse.urlencode(params))
-    json = response.json()['results']
+    json = response.json()['results'] if 'results' in response.json() else response.json()
 
     fresh_addresses = []
 
     for value in json:
         fresh_addresses.append(value['DPA'])
     return fresh_addresses
+
+def get_building_from_toid(toid):
+    """Get building results from TOID term thru UPRN"""
+
+    uprn = 0
+    api_url_1 = f"https://api.os.uk/search/links/v1/identifierTypes/TOID/{toid}?"
+    params = {'key': os.environ.get("os_api_key")}
+    response = requests.get(api_url_1 + urllib.parse.urlencode(params))
+    json_uprn = response.json()
+    if 'correlations' in json_uprn: 
+        uprn = int(json_uprn['correlations'][0]['correlatedIdentifiers'][0]['identifier'])
+    else:
+        uprn = 0
+    if uprn > 0:
+        api_url_2  = "https://api.os.uk/search/places/v1/uprn?"
+        params = {"uprn": uprn, 'key': os.environ.get("os_api_key")}
+        response = requests.get(api_url_2 + urllib.parse.urlencode(params))
+        json = response.json()['results'] if 'results' in response.json() else response.json()
+        fresh_addresses = []
+        for value in json:
+            fresh_addresses.append(value['DPA'])
+        return fresh_addresses
+    else:
+        json = [{'message': 'no uprn found'}]
```

Comment:
It's convention in Rest APIs to not return custom error messages like this. As the successful outcome is an array of places, an empty array is enough for now to indicate that nothing was found.

Like in the get_places_from_search method, the empty array can be defined higher up the method, and then returned at the end (either populated or not)

Reply:
I actually had no idea, good to know!